### PR TITLE
Add debug log for SynchronizedCollection in CI runs for database writes with outdated data

### PR DESCRIPTION
**Pull request by sebbi08**
# Readiness checklist

- [ ] I added/updated tests.
- [x] I ensured that the PR title is good enough for the changelog.
- [x] I labeled the PR.
- [x] I self-reviewed the PR.

### CODE REVIEW

File: packages/transport/src/modules/sync/SynchronizedCollection.ts [hunk 1]
```diff
@@ -87,6 +88,16 @@ export class SynchronizedCollection implements IDatabaseCollection {
 
         const newObjectJson = newObject.toJSON();
 
+        if (this.debugMode) {
+            const oldDocUpdated = Serializable.fromUnknown(await this.parent.read(newObject.id.toString()));
+
+            const readDiff = jsonpatch.compare(oldDocUpdated.toJSON(), oldObject.toJSON());
+            if (readDiff.length > 0) {
+                // eslint-disable-next-line no-console
+                console.error(`Object has been changed since last read.\n${new Error().stack}\n${JSON.stringify(readDiff, null, 2)}`);
```
Comment (jkoenig134) [inline packages/transport/src/modules/sync/SynchronizedCollection.ts:97]: I'd like to give **way more** context on what's wrong here.

Also maybe we can find out that we are in a test any only log then? B/c I think the debugMode flag is also set in the connector, we then don't want to have this kind of log there.

Comment (jkoenig134) [reply packages/transport/src/modules/sync/SynchronizedCollection.ts:97]: from my research one possibility would be to use `typeof globalThis.it === 'function';` to see if you are in a testing environment.

Comment (jkoenig134) [reply packages/transport/src/modules/sync/SynchronizedCollection.ts:97]: great idea. but we cannot access process.env in prod code to be browser compatible ..

Comment (sebbi08) [reply packages/transport/src/modules/sync/SynchronizedCollection.ts]: I just tested this `if (typeof process === "object" && process.env.CI === "true")` and it works in node and browser.

I just need to test if the bundling in the app has any problem with it

File: packages/transport/src/modules/sync/SynchronizedCollection.ts [hunk 2]
```diff
@@ -15,7 +15,8 @@ export class SynchronizedCollection implements IDatabaseCollection {
     public constructor(
         private readonly parent: IDatabaseCollection,
         private readonly datawalletVersion: number,
-        private readonly datawalletModifications?: IDatabaseCollection
+        private readonly datawalletModifications?: IDatabaseCollection,
+        private readonly debugMode?: boolean
```
Comment (jkoenig134) [inline packages/transport/src/modules/sync/SynchronizedCollection.ts:19]: this is unused now

File: packages/transport/src/modules/sync/SynchronizedCollection.ts [hunk 3]
```diff
@@ -87,6 +88,23 @@ export class SynchronizedCollection implements IDatabaseCollection {
 
         const newObjectJson = newObject.toJSON();
 
+        if (typeof process === "object" && process.env.CI === "true") {
```
Comment (jkoenig134) [inline packages/transport/src/modules/sync/SynchronizedCollection.ts:91]:
```suggestion
        if (typeof globalThis.process === "object" && globalThis.process.env.CI) {
```

two things:
- used globalThis to be clear about where process should be defined or not
- don't exactly pin CI (we don't care what's inside this var)

File: packages/transport/src/modules/accounts/AccountController.ts [hunk 4]
```diff
@@ -438,10 +438,10 @@ export class AccountController {
     public async getSynchronizedCollection(collectionName: string): Promise<SynchronizedCollection> {
         const collection = await this.db.getCollection(collectionName);
         if (!this.config.datawalletEnabled) {
-            return new SynchronizedCollection(collection, this.config.supportedDatawalletVersion);
+            return new SynchronizedCollection(collection, this.config.supportedDatawalletVersion, undefined, this.config.debug);
         }
 
-        return new SynchronizedCollection(collection, this.config.supportedDatawalletVersion, this.unpushedDatawalletModifications);
+        return new SynchronizedCollection(collection, this.config.supportedDatawalletVersion, this.unpushedDatawalletModifications, this.config.debug);
```
Comment (jkoenig134) [inline packages/transport/src/modules/accounts/AccountController.ts:444]: you should also undo that.

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**